javaslang/src/main/java/javaslang/Lazy.java

```diff
@@ -101,7 +101,7 @@ private Lazy(Supplier<? extends T> supplier) {
      */
     public static <T> Lazy<Seq<T>> sequence(Iterable<? extends Lazy<? extends T>> values) {
         Objects.requireNonNull(values, "values is null");
-        return Lazy.of(() -> List.ofAll(values).map(Lazy::get));
+        return Lazy.of(() -> Stream.ofAll(values).map(Lazy::get));
```

Comment:
Please use Vector.ofAll() instead of Stream.ofAll(). See an explanation below (at the Option.sequence() change).